src/storage-client/src/sink.rs

```diff
@@ -324,9 +361,17 @@ pub async fn build_kafka(
         KafkaConsistencyConfig::Progress { topic } => ensure_kafka_topic(
             &client,
             topic,
-            1,
-            connection.replication_factor,
-            KafkaSinkConnectionRetention::default(),
+            TopicConfig {
+                partition_count: 1,
+                // TODO: introduce and use `PROGRESS TOPIC REPLICATION FACTOR`
+                // on Kafka connections.
+                replication_factor: -1,
+                retention: KafkaSinkConnectionRetention {
+                    duration: Some(-1),
+                    bytes: Some(-1),
+                },
```

Comment:
Explicitly disabling retention to override whatever broker default retention policy might exist seemed worthwhile.